src/Main.hs

```diff
@@ -99,14 +105,17 @@ stylishHaskell sa = do
 
         else do
             conf <- loadConfig verbose' (saConfig sa)
+            filesR <- case (saRecursive sa) of
+              True -> findFiles (saVerbose sa) (saFiles sa)
+              _    -> return $ saFiles sa
             let steps = configSteps conf
             forM_ steps $ \s -> verbose' $ "Enabled " ++ stepName s ++ " step"
             verbose' $ "Extra language extensions: " ++
                 show (configLanguageExtensions conf)
-            mapM_ (file sa conf) files'
+            mapM_ (file sa conf) $ files' filesR
   where
     verbose' = makeVerbose (saVerbose sa)
-    files'   = if null (saFiles sa) then [Nothing] else map Just (saFiles sa)
+    files' x = if null x then [Nothing] else map Just x
```

Comment:
We probably want to change this... I think if you would run something like `stylish-haskell -r dir-with-no-haskell-files/`, `stylish-haskell` would end up with no files here and wait for `stdin`, which is not what the user meant.  We should change the code here to use `[Nothing]` if:
1. `-r` is not set (alternatively, we could use `.` in that case)
2. No file paths are explicitly given.

Reply:
I am puzzling on this one. Please confirm whether my interpretation is correct:
When `-r` is set then `stdin` shall not be considered at all.

I would not involve using `.`.

Reply:
Correct: when `-r` is set `stdin` should not be considered.